Feature/variation api

mgrbyte opened the pull request.

Commit:
Added schema definitions for variations.

Commit:
Variations and importer.

Commit:
Added variation name pattern to seed data.

(Regexp definition from Paul Davis)

Commit:
Split importer.

Commit:
Implemented variation import.

- Added event types to seed data.
- Implement variation import with constant provenance.

Commit:
Added specs for variation attributes.

Commit:
Merge branch 'feature/variation-importer' into feature/variation-api

Commit:
Finish variation import.

- Tested with latest export data.
- Remove dropping of names when they're identical to ID.
- Simplified code in schema install.
- Extracted people import into a fn for re-use outside of schema installation.

Commit:
Variations cannot be suppressed.

Commit:
Added schema attributes and seed-data for variations.

sibyl229 reviewed (approved):
Yay variation is coming 👍 Just let me know when it needs UI work

Commit:
All tests passing.

Commit:
Used batch info endpoint in creation test.

- Check that the info endpoint produces the expected response.

Commit:
Re-work entity creation.

- Added tests for new variations.
- Extended tests for new gene (conflict)
- Generically handle spec conforming.
- Fixed issue with error handling in case of unique-conflict.

Commit:
Made conform data util function always return a 2-tuple.

Commit:
Generate valid names in test.

Commit:
Removed printing.

Commit:
Fixed errors in variation update handler.

Commit:
Added tests for variation.

Commit:
Renamed file to reflect multi-entity-type testing.

Commit:
Revert "Made conform data util function always return a 2-tuple."

This reverts commit ff7419dd479bc7cbb55811fba368c4bfd831e251.

Commit:
Added tests, batch sub-modules and generic matching.

Commit:
Added tests for batch variation update.

Made it so conformer function needs passing to all uses of
`wormbase.names.batch.generic/batcher`.

Commit:
Add more tests for batch api.

- Made settings swagger responses in capi resources less verbose.
- Fixed issue with response spec for changing status.

Commit:
Fixed error printing in test.
